core/src/main/java/com/datastrato/gravitino/authorization/AdminManager.java

```diff
@@ -37,37 +37,54 @@ class AdminManager {
+  private final RoleManager roleManager;
 
-  AdminManager(EntityStore store, IdGenerator idGenerator, Config config) {
+  AdminManager(EntityStore store, IdGenerator idGenerator, Config config, RoleManager roleManager) {
     this.store = store;
     this.idGenerator = idGenerator;
     this.serviceAdmins = config.get(Configs.SERVICE_ADMINS);
+    this.roleManager = roleManager;
+
+    updateSystemMetalakeUsers();
   }
 
   User addMetalakeAdmin(String user) {
-
-    UserEntity userEntity =
-        UserEntity.builder()
-            .withId(idGenerator.nextId())
-            .withName(user)
-            .withNamespace(
-                Namespace.of(
-                    Entity.SYSTEM_METALAKE_RESERVED_NAME,
-                    Entity.AUTHORIZATION_CATALOG_NAME,
-                    Entity.ADMIN_SCHEMA_NAME))
-            .withRoleNames(Lists.newArrayList())
-            .withAuditInfo(
-                AuditInfo.builder()
-                    .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
-                    .withCreateTime(Instant.now())
-                    .build())
-            .build();
     try {
-      store.put(userEntity, false /* overwritten */);
-      return userEntity;
+      if (isServiceAdmin(user)) {
+        if (hasCreateMetalakeRole(user)) {
+          throw new EntityAlreadyExistsException(
+              "Metalake admin has added the metalake create role");
+        }
+        return updateUserEntity(
+            user,
+            Lists.newArrayList(
+                Entity.METALAKE_CREATE_ROLE, Entity.SYSTEM_METALAKE_MANAGE_USER_ROLE));
+      } else {
+        long roleId =
+            roleManager
+                .getRole(Entity.SYSTEM_METALAKE_RESERVED_NAME, Entity.METALAKE_CREATE_ROLE)
+                .id();
+        UserEntity userEntity =
+            UserEntity.builder()
+                .withId(idGenerator.nextId())
+                .withName(user)
+                .withNamespace(
+                    AuthorizationUtils.ofUserNamespace(Entity.SYSTEM_METALAKE_RESERVED_NAME))
+                .withRoleNames(Lists.newArrayList(Entity.METALAKE_CREATE_ROLE))
+                .withRoleIds(Lists.newArrayList(roleId))
+                .withAuditInfo(
+                    AuditInfo.builder()
+                        .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                        .withCreateTime(Instant.now())
+                        .build())
+                .build();
+
+        store.put(userEntity, false /* overwritten */);
+        return userEntity;
+      }
     } catch (EntityAlreadyExistsException e) {
-      LOG.warn("User {} in the metalake admin already exists", user, e);
-      throw new UserAlreadyExistsException("User %s in the metalake admin already exists", user);
+      LOG.warn("The metalake admin {} has been added before", user, e);
+      throw new RoleAlreadyExistsException("The metalake admin %s has been added before", user);
```

Comment:
Why do you throw "EntityAlreadyExistsException" and convert to "RoleAlreadyExistsException", why don't you directly throw "RoleAlreadyExistsException"?